pkg/config/config.go

```diff
@@ -136,171 +155,208 @@ type UpdateSetting struct {
+	Kubeconfig            string        `yaml:"kubeconfig"`
 }
 
 func (eventType EventType) String() string {
 	return string(eventType)
 }
 
-// NewCommunicationsConfig return new communication config object
-func NewCommunicationsConfig() (*Communications, error) {
-	configPath := os.Getenv("CONFIG_PATH")
-	commCfgFilePath := filepath.Join(configPath, CommunicationConfigFileName)
+// PathsGetter returns the list of absolute paths to the config files.
+type PathsGetter func() []string
 
-	k := koanf.New(communicationConfigDelimiter)
+// LoadWithDefaults loads new configuration from files and environment variables.
+func LoadWithDefaults(getCfgPaths PathsGetter) (*Config, []string, error) {
+	configPaths := getCfgPaths()
+	k := koanf.New(configDelimiter)
 
-	// Load base YAML config.
-	if err := k.Load(file.Provider(filepath.Clean(commCfgFilePath)), koanfyaml.Parser()); err != nil {
-		return nil, err
+	// load default settings
+	if err := k.Load(rawbytes.Provider(defaultConfiguration), koanfyaml.Parser()); err != nil {
+		return nil, nil, fmt.Errorf("while loading default configuration: %w", err)
 	}
 
-	// Load environment variables and merge into the loaded config.
+	// merge with user conf files
+	for _, path := range configPaths {
+		if err := k.Load(file.Provider(filepath.Clean(path)), koanfyaml.Parser()); err != nil {
+			return nil, nil, err
+		}
+	}
+
+	// LoadWithDefaults environment variables and merge into the loaded config.
 	err := k.Load(env.Provider(
-		communicationEnvVariablePrefix,
-		communicationConfigDelimiter,
-		normalizeCommunicationConfigEnvName,
+		configEnvVariablePrefix,
+		configDelimiter,
+		normalizeConfigEnvName,
 	), nil)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	var cfg Communications
+	var cfg Config
 	err = k.UnmarshalWithConf("", &cfg, koanf.UnmarshalConf{Tag: "yaml"})
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &cfg, nil
+	return &cfg, configPaths, nil
 }
 
-func normalizeCommunicationConfigEnvName(name string) string {
-	name = strings.ToLower(name)
-	return strings.ReplaceAll(name, "_", ".")
-}
-
-// Load loads new configuration from file.
-func Load(dir string) (*Config, error) {
-	resCfgFilePath := filepath.Join(dir, ResourceConfigFileName)
-	rawCfg, err := os.ReadFile(filepath.Clean(resCfgFilePath))
-	if err != nil {
-		return nil, err
+// FromEnvOrFlag resolves and returns paths for config files.
+// 1. from 'CONFIG_PATH' env variable (backward compatible). If not found, then:
+// 2. from 'BOTKUBE_CONFIG_PATHS' env variable. If not found, then:
+// 3. from '--config' flag.
+func FromEnvOrFlag() []string {
+	// old style where `comm_config.yaml` and `resource_config.yaml` were used.
+	oldCfgPath := os.Getenv("CONFIG_PATH")
+	if oldCfgPath != "" {
+		return []string{
+			filepath.Join(oldCfgPath, "comm_config.yaml"),
+			filepath.Join(oldCfgPath, "resource_config.yaml"),
+			filepath.Join(oldCfgPath, "analytics.yaml"),
+		}
 	}
```

Comment:
TBH I'm not sure if we need to maintain the backward compatibility here, if we have already some breaking changes in the config itself. Do you have any specific case in mind where we run new image with old Helm chart? 🤔 If not, then I'd rather remove it. WDYT?

Reply:
It was done before I added breaking changes in other places, I will remove it 👍